pkg/collector/helpers_test.go

```diff
@@ -0,0 +1,87 @@
+package collector
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"regexp"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/alphagov/paas-rds-metric-collector/pkg/metrics"
+)
+
+// openMultipleDBConns opens as many connections as specified by
+// count using the given driver and url.
+func openMultipleDBConns(ctx context.Context, count int, driver, url string) (err error, execQuery func(string)) {
+	var dbConns []*sql.DB
+
+	go func() {
+		select {
+		case <-ctx.Done():
+			for _, c := range dbConns {
+				c.Close()
+			}
+		}
+	}()
+
+	execQuery = func(q string) {
+		for _, c := range dbConns {
+			go c.ExecContext(ctx, q)
+		}
+	}
+
+	for i := 0; i < count; i++ {
+		dbConn, err := sql.Open(driver, url)
+		if err != nil {
+			break
+		}
+		err = dbConn.Ping()
+		if err != nil {
+			break
+		}
+		dbConns = append(dbConns, dbConn)
+	}
+	return err, execQuery
+}
+
+func getMetricByKey(collectedMetrics []metrics.Metric, key string) *metrics.Metric {
+	for _, metric := range collectedMetrics {
+		if metric.Key == key {
+			return &metric
+		}
+	}
+	return nil
```

Comment:
Opinion: this, given the name of the method, is an error condition, not a `return nil`.